src/main/java/com/owncloud/android/utils/DrawerMenuUtil.java

```diff
@@ -42,31 +42,26 @@ private DrawerMenuUtil() {
 
     public static void filterSearchMenuItems(Menu menu,
                                              User user,
-                                             Resources resources,
-                                             boolean hasSearchSupport) {
-        if (!user.isAnonymous() && !hasSearchSupport) {
+                                             Resources resources) {
+        if (user.isAnonymous()) {
             filterMenuItems(menu, R.id.nav_photos, R.id.nav_favorites, R.id.nav_videos);
         }
 
-        if (hasSearchSupport) {
-            if (!resources.getBoolean(R.bool.recently_added_enabled)) {
-                menu.removeItem(R.id.nav_recently_added);
-            }
+        if (!resources.getBoolean(R.bool.recently_added_enabled)) {
+            menu.removeItem(R.id.nav_recently_added);
+        }
 
-            if (!resources.getBoolean(R.bool.recently_modified_enabled)) {
-                menu.removeItem(R.id.nav_recently_modified);
-            }
+        if (!resources.getBoolean(R.bool.recently_modified_enabled)) {
+            menu.removeItem(R.id.nav_recently_modified);
+        }
 
-            if (!resources.getBoolean(R.bool.videos_enabled)) {
-                menu.removeItem(R.id.nav_videos);
-            }
-        } else if (!user.isAnonymous()) {
-            filterMenuItems(menu, R.id.nav_recently_added, R.id.nav_recently_modified, R.id.nav_videos);
+        if (!resources.getBoolean(R.bool.videos_enabled)) {
+            menu.removeItem(R.id.nav_videos);
         }
     }
 
     public static void filterTrashbinMenuItem(Menu menu, User user, @Nullable OCCapability capability) {
-        if (!user.isAnonymous() &&
+        if (!user.isAnonymous() ||
```

Comment:
sure?

Reply:
wouldn't it have to be 
```suggestion
        if (user.isAnonymous() ||
```

As in remove menu item in case user _is_ anonymous?

Reply:
I guess we can just remove this.
It was previously needed as we had the check based on account, as we needed to check if the server (which is identified via account) is capable of search.

But as nowadays every supported NC server has search support, this is always true and thus we can remove also this check.
(If a user is anonymous it means that the app is not logged in, so it should not show this screen at all…)

I will do it on Tuesday.

Reply:
anonymous user is the old `account == null`, which in turn was ususally a retrofitted workaround for a NPE bug